tests/test_func.py

```diff
@@ -77,16 +79,18 @@ def _test(self, tocheck):
         except Exception as ex:
             # need finalization to restore a correct state
             self.linter.reporter.finalize()
-            ex.file = tocheck
+            ex.file = tocheck  # type: ignore
```

Comment:
Does this part of the code actually get used? I couldn't get it to trigger when running the tests with a debugger.

If not, can we remove it?
If yes, what is the exact type of `ex` and can we do something to avoid these two `type: ignore`'s?

Reply:
I think this is only to get an idea of what is failing when your modifying pylint and failing at it. We can sanitize the code:

```python
        except Exception as ex:
            msg = f"Exception: {ex} ({exception_str} while checking {to_check}))
            raise Exception(msg) from ex
        finally:
            self.linter.reporter.finalize()
```